### PR TITLE
Java design pattern creational singleton

### PR DESCRIPTION
**Pull request by Bernard-McGeever**
### Description

I have added a short entry describing a creational design pattern; the singleton. This includes a short definition, UML diagram (added to the media folder) and code example (written in Java.)

I wanted to write a series of short definitions on the creational patterns including:
- Singleton
- Builder
- Prototype
- Factory
- Abstract Factory

This is to aid my own education.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by yangc95** [comment]
Hi @Bernard-McGeever,

I see this is your first contribution to Codecademy Docs and I wanted to welcome you here! I'm happy to be your first reviewer on this PR. 

  I’m going to make some suggestions on your entry. **Please tag me** once you've made the appropriate changes based on my suggestions and comments so we can re-evaluate. After I approve the changes, we’ll pass this along to your second reviewer.  

Thanks!

File: content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md
Comment (yangc95) [inline content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md:24]: In terms of pedagogy, it may be beneficial to explain the code before the actual code snippet.

While comments within code snippets are convenient, let's try moving some larger chunks of comments above the code.

File: content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md
Comment (yangc95) [inline content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md:38]: All comments will start out capitalized as per our style guide, ty!

File: content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md
Comment (yangc95) [inline content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md:3]:
```suggestion
Description: 'The singleton pattern ensures a class has only one object instantiated during the programs lifecycle.'
```

File: content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md
Comment (yangc95) [inline content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md:17]:
```suggestion
The singleton pattern ensures a class has only one object instantiated during the program's lifecycle. It is used to guarantee the control of a resource through its implementation.
```

**Comment by Bernard-McGeever** [comment]
On Friday I received an email from Codecademy advertising a new course produced by yourselves and think it is more relevant to this entry. Design patterns should are language agnostic and therefore cant see why learning java should be tagged in the metadata. I think the course _software-design-principles_ is more relevant. I didn't want to change during the PR process, simply because I don't know if replacing line would break anything. There doesn't seem to be a list of acceptable CatalogContent values in the repo.

Please advise?

**Comment by Bernard-McGeever** [comment]
Hi @yangc95,

Changes have been made following your advice :)

Please let me know your thoughts.

**Review by yangc95 — APPROVED** [review]

**Comment by SSwiniarski** [comment]
@Bernard-McGeever, I'll be doing the second review.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
I've reviewed this and just have a couple of suggestions. Let us know when they're addressed.

**Review by SSwiniarski — DISMISSED** [review]
@Bernard-McGeever, I've reviewed this and suggested a couple of changes. Let us know when they're addressed.

**Comment by yangc95** [comment]
> On Friday I received an email from Codecademy advertising a new course produced by yourselves and think it is more relevant to this entry. Design patterns should are language agnostic and therefore cant see why learning java should be tagged in the metadata. I think the course _software-design-principles_ is more relevant. I didn't want to change during the PR process, simply because I don't know if replacing line would break anything. There doesn't seem to be a list of acceptable CatalogContent values in the repo.
> 
> 
> 
> Please advise?

Also wanted to address this; currently working on adding a Content Catalog md to the documentation directory, so make sure to look out for that once it’s up.

Add any suggestions for slugs there. Thanks for the feedback!

File: content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md
Comment (Bernard-McGeever) [reply content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md:21]: Apologies for this one. I knew your preference but had forgotten to update articles previously written.

**Comment by Bernard-McGeever** [comment]
@SSwiniarski - Addressed your comments. Thanks again for the review!

**Comment by sonnynomnom** [comment]
@Bernard-McGeever It's live here: https://www.codecademy.com/resources/docs/general/what-is-the-singleton-pattern

I also made more updates: https://github.com/Codecademy/docs/commit/111531ad46ad3f13cfbf04c04592c61f90b5e5d1
- Changed the name from "What is the Singleton Pattern?" to just "Singleton Pattern" (We no longer do "What is" titles anymore)
- Updated the image link.
- Added single ticks around `getInstance()` and `synchronized`.
- Change four spaces to two spaces for indentation.
- Deleted the periods in spaces.

@christian-dinh @SSwiniarski

**Comment by sonnynomnom** [comment]
One last thing @Bernard-McGeever, make sure your Codecademy profile is linked to your GitHub profile, because right now your profile picture is not showing up!

Screen Shot 2021-11-17 at 5 35 56 PM

**Pull request merged**
